lib/sprite_batch.dart

```diff
@@ -1,67 +1,269 @@
+import 'dart:collection';
 import 'dart:ui';
 
 import 'package:flutter/foundation.dart';
 
+import 'assets/images.dart';
 import 'extensions/vector2.dart';
 import 'flame.dart';
+import 'game/game.dart';
 
-/// sprite atlas with an image and a set of rects and transforms
+extension SpriteBatchExtension on Game {
+  /// Utility method to load and cache the image for a [SpriteBatch] based on its options
+  Future<SpriteBatch> loadSpriteBatch(
+    String path, {
+    Color defaultColor = const Color(0x00000000),
+    BlendMode defaultBlendMode = BlendMode.srcOver,
+    RSTransform defaultTransform,
+  }) {
+    return SpriteBatch.load(
+      path,
+      defaultColor: defaultColor,
+      defaultBlendMode: defaultBlendMode,
+      defaultTransform: defaultTransform,
+      images: images,
+    );
+  }
+}
+
+/// A single item in a SpriteBatch.
+///
+/// Holds all the important information of a batch item,
+///
+/// Web currently does not support `Canvas.drawAtlas`, so a BatchItem will
+/// automatically calculate a transform matrix based on the [transform] value, to be
+/// used when rendering on the web. It will initialize a [destination] object
+/// and a [paint] object.
+class BatchItem {
+  /// The source rectangle on the [SpriteBatch.atlas].
+  final Rect source;
+
+  /// The destination rectangle for the Canvas.
+  ///
+  /// It will be transformed by [matrix].
+  final Rect destination;
+
+  /// The transform values for this batch item.
+  final RSTransform transform;
+
+  /// The background color for this batch item.
+  final Color color;
+
+  /// Fallback matrix for the web.
+  ///
+  /// Because `Canvas.drawAtlas` is not supported on the web we also
+  /// build a `Matrix4` based on the [transform] values.
+  final Matrix4 matrix;
+
+  /// Paint object used for the web.
+  final Paint paint;
+
+  BatchItem({
+    @required this.source,
+    @required this.transform,
+    @required this.color,
+  })  : assert(source != null),
+        assert(transform != null),
+        assert(color != null),
+        matrix = Matrix4(
```

Comment:
despite our styleguide rule of one param per line, since this is a 4x4 matrix, this is a place I would personally open an exception and allow it to be grouped 4 per line.
```dart
matrix = Matrix4(
    1, 2, 3, 4,
    5, 6, 7, 8,
    9, 10, 11, 12,
    13, 14, 15, 16,
)
```
something like this.

not sure if it will look prettier though, might be worth a shot just to see

Reply:
Will the formatter not mess with it? Also not sure How we can fit in the `scale` comment then ;(

Reply:
Ah the formatter does not mess with it, never mind. I'll look into making this work with the comment :-)

Reply:
So `flutter format` does mess with it, but I was able to "hack" it by adding a `//` behind each line.

Reply:
I like that!

but if you think the previous is clearer bc of the comment that is good as well

Reply:
I moved the comment to a const variable, IMHO still readable so I'll resolve it as is